### DISCUSSION TITLE
How to have tests on multiple c files?

### DISCUSSION THREAD

**Discussion by stefano-p** [exoticlibraries/libcester#54] [Q&A]
I was evaluating this framework and found that is seems not possible to build multiple c test files. So do I have to put all tests in one single module?

This file alone ( test_all.c) compiles and run correctly. 
### test_all.c
```
#define CESTER_NO_MAIN
#include <exotic/cester.h>

CESTER_TEST(test1, test_instance,
            cester_assert_equal(NULL, NULL);
)

CESTER_TEST(test_two, test_instance,
    cester_assert_ptr_equal(test_instance, NULL);    
)

CESTER_TEST(test_three, test_instance,
    cester_assert_short_eq(1, 1);    
)

CESTER_BODY(
int main(int argc, char** argv) {
        return CESTER_RUN_ALL_TESTS(argc, argv);
}
)

```
But as I add a new one (test_my_module1.c) I get some strange linker errors saying that there are:

- multiple definition of `superTestInstance'; 
- multiple definition of `default_color'; 
- multiple definition of `hConsole';

I also had to define CESTER_NO_MAIN and CESTER_NO_SIGNAL in order to lower the number of "multiple definition" errors.

### test_my_module1.c
```
#define CESTER_NO_MAIN
#define CESTER_NO_SIGNAL
#include <exotic/cester.h>

CESTER_TEST(test4, test_instance,
            cester_assert_equal(NULL, NULL);
)
```
Is it possible to spread tests accross multiple c files? How?

**Comment by Thecarisma** [comment]
Hello Stefano

It currently not possible to compile multiple source file. I have not have a scenario to compile multiple test file. You are the first to point that out, I'll check today to try compile multiple source, if it possible I'll add the feature.

I hope by multiple source compilation you meant.

```
gcc test_my_module1.c test_all.c -I
```

**Reply by stefano-p** [reply]
Yes @Thecarisma, that is what I meant. It would be a very important feature that all other framework I have tested so far already have. All my projects would require separate c test files, each one dedicated to a specific module. Putting all tests in one c file would not be good, for clarity and maintenance reasons. In principle, each module would have its associated test file. Then all tests are run in one shot.

**Reply by OMaghiarIMG** [reply]
Hello, is there any progress for supporting multiple test files?

**Reply by shanebishop** [reply]
@Thecarisma, I agree with @stefano-p that test frameworks should support breaking tests out across multiple files, in the same way that source code can be broken across multiple files. I created https://github.com/exoticlibraries/libcester/issues/71 from this question.
